### ISSUE TITLE
TDS Fee Bugs Reported 12/6/22 (On PROD)

### ISSUE THREAD

**Issue by TracyLinder** [cityofaustin/atd-data-tech#10952]
### Fee bugs that need to be scoped and fixed:
`TDS # 1 AMANDA Priority`
**Customer:** Daniil 

Acceptance Criteria 
- Need to ensure all fees attached in the excel file below have correct fee amounts updated
- Need to add a Fee Procedure in the Insert and Run Procedure menu for "ATD Tech Surcharge" that inserts "ATD Tech Surcharge ROW" fees as a line item and 4% of any existing fee as the fee amount. 

[https://app.zenhub.com/files/140626918/465eb93c-1959-46d8-8ee6-db043bee4c8b/download](https://app.zenhub.com/files/140626918/465eb93c-1959-46d8-8ee6-db043bee4c8b/download)

1. The ATD Tech Surcharge is not always inserting with all TDS fees
2. The fee run procedure doesn't work on certain fees. 

**Notes for Scoping:**

- [ ] We need a list of the fees that aren't inserting with the surcharge and what folders they are on.
- [ ] We need to know which fee procedures aren't working, on which folder(s), what the fee is and the amount. Does the fee exist in the DB yet? If not please inform Tracy (so I can worry about it).

**Comment by TracyLinder** [comment]
Hi @vschandramouli can you please schedule time with Daniil to scope these bugs? I put scoping notes in the description. Thank you.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  @vschandramouli  
1. From the given excel sheet all the RSNs are of C, C8, SP, ZC. So, I am going to add ATD Tech Surcharge ROW (50174) to Run procedure for all these foldertypes.
2. In the given excel sheet, which is column is the latest fee amount?

**Comment by SubhaShroff** [comment]
@Nadin-Nader  @vschandramouli 
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/SUBHAS~1/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/SUBHAS~1/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">



83416 | ATD-SP-Non-TIA Mitigation Fee   < 2,000 trips per day    -------($660/2010) |   |  | As per the code, it must be   inserting $2810 in PROD. But the two records show different amount. May be   both are entered manually. Note: in DEV/TEST-$3560. But for the given RSNs it   shows two different amounts $660/$2010. Please confirm if it’s a variable   fee? If so What is the formula? Or confirm on the expected amount.
-- | -- | -- | -- | --




</body>

</html>

**Comment by SubhaShroff** [comment]
And to add to the above, following fee amounts are different only in PROD. I have texted George if he can fix those:<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/SUBHAS~1/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/SUBHAS~1/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">



83497 | ATD-SP-Other-Revision,D,A,B,CD,DS,BS,T,W/R,Concept\ | 3160 | This is $3160 in DEV/TEST, but   different in PROD. Will need George's help with this.
-- | -- | -- | --
83475 | ATD-SP-Site Plan Consolidated | 6320 | looks correct in TEST but PROD   needs correction
83472 | ATD-SP-Administrative Extension | 790 | in DEV/TEST-Its correct 790.   PROD its 700.
83478 | ATD-Final Plat | 2370 | in DEV/TEST its 2370. PROD - its   2110
83486 | ATD-ZON-PUD Amendment | 3160 | in DEV/TEST its 3160.   PROD - 2930
83488 | ATD-ZON Review <.25 Acres | 1580 | in DEV/TEST its 1580.  PROD - 1410
83489 | ATD-ZON-Review <.5 Acres | 1580 | in DEV/TEST its 1580.  PROD - 1410
83490 | ATD-ZON-Review <1 Acres | 1580 | in DEV/TEST its 1580.  PROD - 1410
83499 | ATD-SUB-Project Assessment (C8) | 1580 | C8 - in DEV/TEST its 1580.  PROD - 1050



</body>

</html>

**Comment by SubhaShroff** [comment]
@Nadin-Nader  SCTASK0272020 is submitted DEV/TEST deployment of this ticket.

**Comment by SubhaShroff** [comment]
@Nadin-Nader Tested the deployment, the amounts look good. But, though ATD tech surcharge ROW displays in the menu for SP, ZC, C, C8. It doesn't calculate 4% for all fees. I will investigate about it.

**Comment by Nadin-Nader** [comment]
@SubhaShroff here is Daniil's response to your email:

> Nadin,
> 
> This fee will be a general fee which is manually added by our division once Transportation Development Assessment is required. It will not be added during the DSD intake. 
> 
> The fee amount is $1,580.00 (it used to be $1,410.00).
> 
> I hope this answers the question.
> 
> Sincerely,
> 
>  
> Daniil Kontsevoy 
> Project Coordinator: Central Area
> Transportation Development Services Division
> Austin Transportation Department
> Direct (512) 978-1561 | Main (512) 974-1150

**Comment by SubhaShroff** [comment]
@Nadin-Nader  SCTASK0273658 is submitted for DEV/TEST deployment - that will fix the if ATD tech surcharge is manually added it will calculate the 4% for the added unbilled fees(mentioned in this ticket.)

**Comment by SubhaShroff** [comment]
@Nadin-Nader  Tested the above deployment and it needed another commit - [0ef822df87d3](http://coaamandadevops02.austintexas.gov/R5:0ef822df87d3ebc4af16c590007288eb1c61119e). Please approve it when you are back. Thanks.

**Comment by Nadin-Nader** [comment]
The commit has been approved. Ready for the 4% commit to got to TEST.

**Comment by TracyLinder** [comment]
### Email Thread w/ DSD

**From Jesse G. w/ DSD:**

Hi, I work with DSD IT as a tester and we have an incident in our queue about the ATD SP fees haven’t been updated from last year. We were informed that the changes are already in the TEST environment minus the 4% surcharge so we asked the SP staff to test in TEST.

 

They reported that “fee code (83424/83425) Construction Plans and fee code (83499) Preliminary Cases are wrong in test. Everything else looked fine.”

 

Is this something you could look into? Should this INC be transferred to ATD?

 

For contact you could reach out to Zachary Whitaker or Eric Wood who did the testing.

 

Thank You,

Jesse R Gomez



----
**Nate's Response:**

Jesse,

 

Hope you are well. Our developer was working on this and got everything to TEST minus the 4% surcharge. Since we are short staffed the fix for this and the fees wont get sent through to PROD until Tuesday or Wednesday.

Please be aware that this is a high pri for us as well. You can go ahead and close the incident (or hold onto it until the changes are in PROD) since we have a card on our Github board for this.

**Comment by Nadin-Nader** [comment]
@SubhaShroff  from Jesse G. w/DSD:

> From: Gomez, Jesse <Jesse.Gomez@austintexas.gov>
> Sent: Thursday, December 29, 2022 10:28 AM
> To: Linder, Tracy <Tracy.Linder@austintexas.gov>; Nader, Nadin <Nadin.Nader@austintexas.gov>
> Cc: Weeks, Joelyn <Joelyn.Weeks@austintexas.gov>; Rodriguez, David [DSD] <David.Rodriguez2@austintexas.gov>
> Subject: INC0252262 FEE Priority - FY23 Fees - ATD SP Fees are Incorrect
> 
> 
> 
> Hi, I work with DSD IT as a tester and we have an incident in our queue about the ATD SP fees haven’t been updated from last year. We were informed that the changes are already in the TEST environment minus the 4% surcharge so we asked the SP staff to test in TEST.
> 
> 
> 
> They reported that “fee code (83424/83425) Construction Plans and fee code (83499) Preliminary Cases are wrong in test. Everything else looked fine.”

**Comment by SubhaShroff** [comment]
@Nadin-Nader As per the latest email from Jesse,  the change is done and submitted DEV/TEST deployment request - SCTASK0275811.

![image](https://user-images.githubusercontent.com/51680715/210436199-54af4a13-8753-4890-8c64-03d70ce1b5f7.png)

**Comment by SubhaShroff** [comment]
@Nadin-Nader  The deployments are done and verified on TEST. Its ready for UAT in cluding the above fee 83477 amount. Thanks.

**Comment by Nadin-Nader** [comment]
@SubhaShroff sent Jesse an email to test this again. I also asked Daniil.
Waiting on Response

**Comment by Nadin-Nader** [comment]
@SubhaShroff UAT results from Daniil:

4% Surcharge in "run" only works on one fee. It should be available for all fees

[8:37 AM] Kontsevoy, Daniil
I noticed that some subdivision (SUB) fees once added are shown without "SUB" prefix and it can be mistaken for SP fee

[8:41 AM] Kontsevoy, Daniil
This fee should be called "Transportation Assessment"

[8:42 AM] Kontsevoy, Daniil
This subdivision fee has an old amount. It should be $3,160.00

[8:45 AM] Kontsevoy, Daniil
These 2 fees are confusing. 
ATD SUB Construction Plan according to the fee schedule is $3,160.00 (not 2,810.00)
ATD SUB Construction Plan Non-Concurrent isn't in a fee schedule 

[8:47 AM] Kontsevoy, Daniil
SUB Commission Approved Variance should be $2,370.00 (not 2,110.00)


[8:48 AM] Kontsevoy, Daniil
Final plat should be $2,370.00 (not 2,110.00)

**Comment by SubhaShroff** [comment]
@Nadin-Nader  SCTASK0277474 is submitted for DEV/TEST deployment changes.

**Comment by SubhaShroff** [comment]
@Nadin-Nader
###   The following fee names are changed as suggested:

1. 83417	ATD-SP-Transportation Assessment
2. 83424	ATD-SUB-Construction Plan
3. 83426	ATD-SUB-Administrative Non-Environmental Variance
4. 83427	ATD-SUB-Commission Approved Non-Enviromental Variance

### And the following fees are updated so that the Surcharge ROW is added when added from Main Menu:
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/SUBHAS~1/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/SUBHAS~1/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">

</head>

<body link="#0563C1" vlink="#954F72">



83422 | ATD-SP-Administrative Waiver/Variance
-- | --
83423 | ATD-SP-Commission Waiver/Variance
83473 | ATD-SP-Administrative Conditional Use
83472 | ATD-SP-Administrative Extension
83474 | ATD-SP-Administrative Small Site Plan Review
83495 | ATD-SP-Commission Conditional Use
83496 | ATD-SP-Commission Extension
83421 | ATD-SP-MUD Consent Agreement
83424 | ATD-Construction Plan
83426 | ATD-Administrative Non-Environmental Variance
83427 | ATD-Commission Approved Non-Enviromental Variance
83417 | ATD-SP-TIA Review 2,000-5,000 trips per day
83425 | ATD-SUB-Construction Plan Non-Concurrent
83475 | ATD-SP-Site Plan Consolidated



</body>

</html>


### For the following fees the amount is updated to 2370$ for all C, C8, SP and ZC:

- 83478 - ATD-Commission Approved Non-Enviromental Variance -2370
- 83427	ATD-SUB-Final Plat

**Comment by Nadin-Nader** [comment]
Sent Jesse, Danill and Zach a message that the fees were good to test out. Waiting on Response

**Comment by Nadin-Nader** [comment]
Found a few more Fees that were not updated during the UAT with Jesse. Daniil also found several fees that were not given to us and not updated. We also found that the 4% surcharge needs to be added to the fees upon the fees being inserted. 
Subha will be fixing the fees found in the UAT with Jesse but we will fix the 4% surcharge and the fees Daniil found on another card.

**Comment by TracyLinder** [comment]
### Sync (Tracy & Nate)

UAT found that the surcharge isn't inserting w/ fee
UAT found several fees that were misnamed or wrong amount
Daniil provided new fees to be updated that weren't included in discovery

**Comment by SubhaShroff** [comment]
@Nadin-Nader 
[5d358024cc24](http://coaamandadevops02.austintexas.gov/R5:5d358024cc248124e7a01c8ae1c1c5396942fb70) - I made the commit to 
VALIDLOOKUP-90266: 83245 fee amount is changed to 3160
DefaultFee_SP_83417:- fee name changed to ATD-SP-Transportation Assessment to display in fee menu

### Yet to find feecodes and folder type for the following fees to add those in the fee tab menu:
Transportation Development Assessment 
Transportation Completeness Check
Site Plan Corrections 
Plat/Subdivision Vacation

**Comment by TracyLinder** [comment]
@vschandramouli please get with Subha to help find the fee codes she's missing. She also needs to know which folder types they go on.

1. Transportation Development Assessment
2. Transportation Completeness Check
3. Site Plan Corrections
4. Plat/Subdivision Vacation

**Comment by SubhaShroff** [comment]
> @Nadin-Nader [5d358024cc24](http://coaamandadevops02.austintexas.gov/R5:5d358024cc248124e7a01c8ae1c1c5396942fb70) - I made the commit to VALIDLOOKUP-90266: 83245 fee amount is changed to 3160 for C, C8, ZC also. DefaultFee_SP_83417:- fee name changed to ATD-SP-Transportation Assessment to display in fee menu
> 
> ### Yet to find feecodes and folder type for the following fees to add those in the fee tab menu. @Nadin-Nader we need the information from the user:
> Transportation Development Assessment Transportation Completeness Check Site Plan Corrections Plat/Subdivision Vacation

@Nadin-Nader  SCTASK0282383 - DEV/TEST deployment request is submitted for the above commit.

**Comment by TracyLinder** [comment]
@vschandramouli can you please assist Subha in finding fee codes for the 2 fees above?

**Comment by SubhaShroff** [comment]
@TracyLinder You have created another ticket for this - #11314 in yesterday's call. I just did quote reply to my comment for the amount part that I committed. I think Vish must be updating his findings in #11314.
CC: @vschandramouli, @Nadin-Nader

**Comment by TracyLinder** [comment]
@SubhaShroff I agree. Thank you for the clarification.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  SCTASK0283334 is submitted to correct the dates on DEV/TEST.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  Th following issues are fixed on DEV/TEST:
1. 83245 fee amount is changed to 3160 for C, C8, ZC also. 
2. For SP:- fee name changed to ATD-SP-Transportation Assessment to display in fee menu
These two failed in UAT and now fixed/ready for UAT. Thanks.

**Comment by Nadin-Nader** [comment]
Daniil has verified and all the changes look good. Please promote to PROD and move on to 11314

**Comment by SubhaShroff** [comment]
@Nadin-Nader SCTASK0285247 is raised for PROD deployment. Thank you.

**Comment by SubhaShroff** [comment]
@Nadin-Nader  Now its PROD deployed.

**Comment by SubhaShroff** [comment]
### Status:
Two weeks ago it was deployed in PROD. Waiting to close.

**Comment by TracyLinder** [comment]
### Closing
Remaining fee fixes are on #11314
